### PR TITLE
HBASE-22743 : ClientUtils for Demo Client classes

### PR DESCRIPTION
**Pull request by virajjasani**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 82 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 406 | master passed |
| +1 | compile | 31 | master passed |
| +1 | checkstyle | 16 | master passed |
| +1 | shadedjars | 350 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 21 | master passed |
| 0 | spotbugs | 53 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 50 | master passed |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 365 | root in the patch failed. |
| +1 | compile | 30 | the patch passed |
| +1 | javac | 30 | the patch passed |
| -1 | checkstyle | 16 | hbase-examples: The patch generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 371 | patch has no errors when building our shaded downstream artifacts. |
| -1 | hadoopcheck | 222 | The patch causes 11 errors with Hadoop v2.8.5. |
| -1 | hadoopcheck | 451 | The patch causes 11 errors with Hadoop v2.9.2. |
| -1 | hadoopcheck | 683 | The patch causes 11 errors with Hadoop v3.1.2. |
| +1 | javadoc | 22 | the patch passed |
| +1 | findbugs | 59 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 32 | hbase-examples in the patch failed. |
| +1 | asflicense | 14 | The patch does not generate ASF License warnings. |
| | | 2640 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/413 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux d48f71ee316d 4.4.0-154-generic #181-Ubuntu SMP Tue Jun 25 05:29:03 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-413/out/precommit/personality/provided.sh |
| git revision | master / 747d000808 |
| Default Java | 1.8.0_181 |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/1/artifact/out/patch-mvninstall-root.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/1/artifact/out/diff-checkstyle-hbase-examples.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/1/artifact/out/patch-javac-2.8.5.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/1/artifact/out/patch-javac-2.9.2.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/1/artifact/out/patch-javac-3.1.2.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/1/artifact/out/patch-unit-hbase-examples.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/1/testReport/ |
| Max. process+thread count | 96 (vs. ulimit of 10000) |
| modules | C: hbase-examples U: hbase-examples |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 111 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| -1 | mvninstall | 416 | root in master failed. |
| +1 | compile | 24 | master passed |
| +1 | checkstyle | 12 | master passed |
| +1 | shadedjars | 294 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 17 | master passed |
| 0 | spotbugs | 43 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 39 | master passed |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 307 | root in the patch failed. |
| +1 | compile | 23 | the patch passed |
| +1 | javac | 23 | the patch passed |
| -1 | checkstyle | 12 | hbase-examples: The patch generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 312 | patch has no errors when building our shaded downstream artifacts. |
| -1 | hadoopcheck | 180 | The patch causes 10 errors with Hadoop v2.8.5. |
| -1 | hadoopcheck | 379 | The patch causes 10 errors with Hadoop v2.9.2. |
| -1 | hadoopcheck | 543 | The patch causes 10 errors with Hadoop v3.1.2. |
| +1 | javadoc | 18 | the patch passed |
| +1 | findbugs | 47 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 26 | hbase-examples in the patch failed. |
| +1 | asflicense | 11 | The patch does not generate ASF License warnings. |
| | | 2313 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/413 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 190cd57d079c 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-413/out/precommit/personality/provided.sh |
| git revision | master / 747d000808 |
| Default Java | 1.8.0_181 |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/2/artifact/out/branch-mvninstall-root.txt |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/2/artifact/out/patch-mvninstall-root.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/2/artifact/out/diff-checkstyle-hbase-examples.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/2/artifact/out/patch-javac-2.8.5.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/2/artifact/out/patch-javac-2.9.2.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/2/artifact/out/patch-javac-3.1.2.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/2/artifact/out/patch-unit-hbase-examples.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/2/testReport/ |
| Max. process+thread count | 96 (vs. ulimit of 10000) |
| modules | C: hbase-examples U: hbase-examples |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 96 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 332 | master passed |
| +1 | compile | 24 | master passed |
| +1 | checkstyle | 15 | master passed |
| +1 | shadedjars | 269 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 19 | master passed |
| 0 | spotbugs | 41 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 38 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 304 | the patch passed |
| +1 | compile | 24 | the patch passed |
| +1 | javac | 24 | the patch passed |
| -1 | checkstyle | 14 | hbase-examples: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 270 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 939 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 18 | the patch passed |
| +1 | findbugs | 46 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 117 | hbase-examples in the patch passed. |
| +1 | asflicense | 12 | The patch does not generate ASF License warnings. |
| | | 2937 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/413 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 7aa7b6634d03 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-413/out/precommit/personality/provided.sh |
| git revision | master / fa77c75848 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/3/artifact/out/diff-checkstyle-hbase-examples.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/3/testReport/ |
| Max. process+thread count | 2268 (vs. ulimit of 10000) |
| modules | C: hbase-examples U: hbase-examples |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 145 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 503 | master passed |
| +1 | compile | 29 | master passed |
| +1 | checkstyle | 16 | master passed |
| +1 | shadedjars | 368 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 22 | master passed |
| 0 | spotbugs | 54 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 51 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 408 | the patch passed |
| +1 | compile | 26 | the patch passed |
| +1 | javac | 26 | the patch passed |
| -1 | checkstyle | 14 | hbase-examples: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 309 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1237 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 20 | the patch passed |
| +1 | findbugs | 55 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 126 | hbase-examples in the patch passed. |
| +1 | asflicense | 13 | The patch does not generate ASF License warnings. |
| | | 3831 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/413 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux f881f429425e 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-413/out/precommit/personality/provided.sh |
| git revision | master / 1cb37f18aa |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/4/artifact/out/diff-checkstyle-hbase-examples.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/4/testReport/ |
| Max. process+thread count | 2530 (vs. ulimit of 10000) |
| modules | C: hbase-examples U: hbase-examples |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-examples/src/main/java/org/apache/hadoop/hbase/util/ClientUtils.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  }
+
+  /**
+   * Helper to translate byte[]'s to UTF8 strings
+   *
+   * @param buf byte array buffer
+   * @return UTF8 decoded string value
+   */
+  public static String utf8(byte[] buf) {
+    try {
```
Comment (openinx) [inline hbase-examples/src/main/java/org/apache/hadoop/hbase/util/ClientUtils.java]: Just use the Bytes.toString ?  I think it have considered the utf8 decode.

```
  public static String toString(final byte [] b) {
```

File: hbase-examples/src/main/java/org/apache/hadoop/hbase/util/ClientUtils.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.hbase.thrift.generated.TCell;
+import org.apache.hadoop.hbase.thrift.generated.TRowResult;
+import org.apache.yetus.audience.InterfaceAudience;
+
+/**
+ * Common Utility class for clients
+ */
+@InterfaceAudience.Private
+public class ClientUtils {
```
Comment (openinx) [inline hbase-examples/src/main/java/org/apache/hadoop/hbase/util/ClientUtils.java]: Make the class to be final ?

File: hbase-examples/src/main/java/org/apache/hadoop/hbase/util/ClientUtils.java (hunk 1 above)
Comment (virajjasani) [reply hbase-examples/src/main/java/org/apache/hadoop/hbase/util/ClientUtils.java]: Done. Please review

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 61 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 371 | master passed |
| +1 | compile | 30 | master passed |
| +1 | checkstyle | 16 | master passed |
| +1 | shadedjars | 336 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 22 | master passed |
| 0 | spotbugs | 52 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 49 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 360 | the patch passed |
| +1 | compile | 28 | the patch passed |
| +1 | javac | 28 | the patch passed |
| +1 | checkstyle | 16 | the patch passed |
| +1 | whitespace | 1 | The patch has no whitespace issues. |
| +1 | shadedjars | 356 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1131 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 22 | the patch passed |
| +1 | findbugs | 59 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 124 | hbase-examples in the patch passed. |
| +1 | asflicense | 12 | The patch does not generate ASF License warnings. |
| | | 3436 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/413 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 864ed7b4825f 4.4.0-154-generic #181-Ubuntu SMP Tue Jun 25 05:29:03 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-413/out/precommit/personality/provided.sh |
| git revision | master / 1cb37f18aa |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/5/testReport/ |
| Max. process+thread count | 2553 (vs. ulimit of 10000) |
| modules | C: hbase-examples U: hbase-examples |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/5/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 185 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 1 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 362 | master passed |
| +1 | compile | 27 | master passed |
| +1 | checkstyle | 15 | master passed |
| +1 | shadedjars | 268 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 18 | master passed |
| 0 | spotbugs | 42 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 39 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 287 | the patch passed |
| +1 | compile | 23 | the patch passed |
| +1 | javac | 23 | the patch passed |
| +1 | checkstyle | 11 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 283 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 930 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 18 | the patch passed |
| +1 | findbugs | 45 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 121 | hbase-examples in the patch passed. |
| +1 | asflicense | 10 | The patch does not generate ASF License warnings. |
| | | 3008 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/413 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux a7964275c848 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-413/out/precommit/personality/provided.sh |
| git revision | master / cf8114a82e |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/6/testReport/ |
| Max. process+thread count | 2465 (vs. ulimit of 10000) |
| modules | C: hbase-examples U: hbase-examples |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/6/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 63 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 301 | master passed |
| +1 | compile | 23 | master passed |
| +1 | checkstyle | 12 | master passed |
| +1 | shadedjars | 257 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 18 | master passed |
| 0 | spotbugs | 38 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 35 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 280 | the patch passed |
| +1 | compile | 23 | the patch passed |
| +1 | javac | 23 | the patch passed |
| +1 | checkstyle | 11 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 257 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 889 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 16 | the patch passed |
| +1 | findbugs | 45 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 114 | hbase-examples in the patch passed. |
| +1 | asflicense | 10 | The patch does not generate ASF License warnings. |
| | | 2715 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/413 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 5d8b9f8629e0 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-413/out/precommit/personality/provided.sh |
| git revision | master / 3318b4bc6d |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/7/testReport/ |
| Max. process+thread count | 2273 (vs. ulimit of 10000) |
| modules | C: hbase-examples U: hbase-examples |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-413/7/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**